### PR TITLE
feat(react-components): filter pois by scene

### PR DESCRIPTION
**Pull request by haakonflatval-cognite**
#### Type of change

Feat  

#### Jira ticket :blue_book:

https://cognitedata.atlassian.net/browse/

## Checklist :ballot_box_with_check:

- [x] I am proud of this feature.
- [ ] I have performed a self-review of my own code.
- [x] I have added PR type (Feat, Bug, Chore, Test, Docs, Style, Refactor) to the PR title.
- [ ] I have manually tested this for different scenarios (different models, formats, devices, browsers).
- [ ] I have commented my code in hard-to-understand areas.
- [ ] I have made corresponding changes to the public documentation.
- [ ] I have added unit and visuals tests to prove that my feature works to the best of my ability.
- [ ] I have refactored the code for readability to the best of my ability.
- [ ] I have checked that my changes do not introduce regressions in the public documentation.
- [ ] I have outlined any known defects / lacking capabilities in the contents of this PR.
- [ ] I have listed any remaining work that should follow this PR in the description or jira/miro/etc.
- [ ] I have added TSDoc to any public facing changes.
- [ ] I have added "developer documentation" in any package README.md that is related / applicable to this PR.
- [ ] I have noted down and am currently tracking any technical debt introduced in this PR.
- [ ] I have thoroughly thought about the architecture of this implementation.
- [ ] I have asked peers to test this feature.

### CODE REVIEW

File: react-components/src/architecture/concrete/pointsOfInterest/PointsOfInterestDomainObject.ts [hunk 1]
```diff
@@ -36,7 +42,7 @@ export class PointsOfInterestDomainObject<PoiIdType> extends VisualDomainObject
         ...this._pointsOfInterest
       ];
       this.notify(Changes.geometry);
-    });
+    }); */
```
Comment (danpriori) [inline react-components/src/architecture/concrete/pointsOfInterest/PointsOfInterestDomainObject.ts:45]: this commented code should be removed or is it intentional to keep it for now and enable it later?
If is that the case, then I would suggest to add a comment as a header to explain when it should be enabled again.

File: react-components/src/architecture/concrete/pointsOfInterest/PointsOfInterestDomainObject.ts [hunk 2]
```diff
@@ -111,6 +117,8 @@ export class PointsOfInterestDomainObject<PoiIdType> extends VisualDomainObject
     const poisToCreate = this._pointsOfInterest.filter(
       (obs) => obs.status === PointsOfInterestStatus.PendingCreation
     );
+
+    console.log('Trying to upsert points: ', poisToCreate);
```
Comment (danpriori) [inline react-components/src/architecture/concrete/pointsOfInterest/PointsOfInterestDomainObject.ts]: can be removed?

File: react-components/src/architecture/concrete/pointsOfInterest/PointsOfInterestDomainObject.ts [hunk 3]
```diff
@@ -164,4 +171,30 @@ export class PointsOfInterestDomainObject<PoiIdType> extends VisualDomainObject
...
+    if (scene === this._currentScene) {
+      return;
+    }
+
+    this._currentScene = scene;
+    this._pointsOfInterest = [];
+
+    if (scene === undefined) {
+      return;
+    }
```
Comment (danpriori) [inline react-components/src/architecture/concrete/pointsOfInterest/PointsOfInterestDomainObject.ts:189]: Is it possible to move this early return along with the first one on the top of we need to have the internal `this._currentScene` and the `this._pointsOfInterest `reset even when the `scene ` is `undefined` and not when the `scene `is equal to `this._currentScene`?

File: react-components/src/architecture/concrete/pointsOfInterest/ads/PointsOfInterestAdsProvider.ts [hunk 4]
```diff
@@ -42,20 +43,12 @@ export class PointsOfInterestAdsProvider implements PointsOfInterestProvider<Ext
   async upsertPointsOfInterest(
     pois: Array<{ id: ExternalId; properties: PointsOfInterestProperties }>
   ): Promise<Array<PointsOfInterestInstance<ExternalId>>> {
+    const upsertRequestData = createUpsertRequestFromPois(pois);
```
Comment (danpriori) [inline react-components/src/architecture/concrete/pointsOfInterest/ads/PointsOfInterestAdsProvider.ts:46]: It's not important for now but can we have a type for the: 
`{ id: ExternalId; properties: PointsOfInterestProperties }` ?

File: react-components/src/architecture/concrete/pointsOfInterest/fdm/PointsOfInterestFdmProvider.ts [hunk 5]
```diff
@@ -26,8 +22,11 @@ export class PointsOfInterestFdmProvider implements PointsOfInterestProvider<Dms
     return await createPointsOfInterestInstances(this._fdmSdk, pois);
   }
 
-  async fetchAllPointsOfInterest(): Promise<Array<PointsOfInterestInstance<DmsUniqueIdentifier>>> {
-    return await fetchPointsOfInterest(this._fdmSdk);
+  async fetchPointsOfInterest(
+    _scene?: DmsUniqueIdentifier
+  ): Promise<Array<PointsOfInterestInstance<DmsUniqueIdentifier>>> {
+    throw Error('Not implemented:  Fdm PoI scene filter');
+    // return await fetchPointsOfInterest(this._fdmSdk);
```
Comment (danpriori) [inline react-components/src/architecture/concrete/pointsOfInterest/fdm/PointsOfInterestFdmProvider.ts:29]: this returning is commented. Is it intentional?

File: react-components/src/components/Architecture/pointsOfInterest/useFilterPointsOfInterest.ts [hunk 6]
```diff
@@ -23,7 +23,7 @@ function filterPoi(poi: PointOfInterest<unknown>, filter: PoiFilter | undefined)
   const lowerCaseContainsFilter = filter.contains.toLowerCase();
 
   return (
-    poi.properties.title?.toLowerCase().includes(lowerCaseContainsFilter) === true ||
+    poi.properties.title?.toLowerCase().includes(lowerCaseContainsFilter) ||
```
Comment (danpriori) [inline react-components/src/components/Architecture/pointsOfInterest/useFilterPointsOfInterest.ts:26]: just a cosmetic suggestion: maybe it is better to move these two statements to a var (before the return) and then return a var result instead.

**Review by danpriori — CHANGES_REQUESTED** [review]
My main concern is the commented code, which is I dont know if they were intentional. The rest is good! :)

File: react-components/src/architecture/concrete/pointsOfInterest/PointsOfInterestDomainObject.ts (hunk 1 above)
Comment (haakonflatval-cognite) [reply react-components/src/architecture/concrete/pointsOfInterest/PointsOfInterestDomainObject.ts:45]: Woops :see_no_evil: I'll remove

File: react-components/src/architecture/concrete/pointsOfInterest/PointsOfInterestDomainObject.ts (hunk 2 above)
Comment (haakonflatval-cognite) [reply react-components/src/architecture/concrete/pointsOfInterest/PointsOfInterestDomainObject.ts]: Hehe, yes

File: react-components/src/architecture/concrete/pointsOfInterest/PointsOfInterestDomainObject.ts (hunk 3 above)
Comment (haakonflatval-cognite) [reply react-components/src/architecture/concrete/pointsOfInterest/PointsOfInterestDomainObject.ts:189]: We need to reset `this._currentScene` and the points when the scene is undefined. So I'd like to keep it here

File: react-components/src/architecture/concrete/pointsOfInterest/ads/PointsOfInterestAdsProvider.ts (hunk 4 above)
Comment (haakonflatval-cognite) [reply react-components/src/architecture/concrete/pointsOfInterest/ads/PointsOfInterestAdsProvider.ts:46]: I think we have one - I'll fix :sweat_smile:

File: react-components/src/architecture/concrete/pointsOfInterest/fdm/PointsOfInterestFdmProvider.ts (hunk 5 above)
Comment (haakonflatval-cognite) [reply react-components/src/architecture/concrete/pointsOfInterest/fdm/PointsOfInterestFdmProvider.ts:29]: Yes, it's intentional.. Although it's a bit ugly. I was hoping we could have the FDM provider ready by Tuesday, but not sure anymore :/

Comment (danpriori) [reply react-components/src/architecture/concrete/pointsOfInterest/fdm/PointsOfInterestFdmProvider.ts:29]: Hmmm ok. It's fine. We have to keep track on this to not let it live that long enough to forget that it still there.

**Review by danpriori — APPROVED** [review]
LGTM

**Pull request merged**